port/common/omrsysinfo_helpers.h

```diff
@@ -0,0 +1,39 @@
+/*******************************************************************************
+ * Copyright (c) 2019, 2019 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+/**
+ * @file
+ * @ingroup Port
+ * @brief System information
+ */
+
+#ifndef SYSINFOHELPERS_H_
+#define SYSINFOHELPERS_H_
+
+#include "omrport.h"
+
+extern void
+getX86CPUID(uint32_t leaf, uint32_t *cpuInfo);
+
+extern intptr_t
+getX86Description(struct OMRPortLibrary *portLibrary, OMRProcessorDesc *desc);
```

Comment:
The corresponding Z `getstfle` function which performs similar operations is in the `port/linuxs390` directory. Should this go into `port/linux386`? It does not look like common code, so to me it is odd that it is currently in `port/common`.

Reply:
I just copied the implementation of OpenJ9's port library,  but I do believe `getX86Description` is used no only in both `win32/omrsysinfo.c` as well as in `unix/omrsysinfo.c`. Not too sure why it was done like that in OpenJ9, but I'm assuming that we might get a compilation error if we add `getX86Description` to `port/linux386` (even if the call is protected by an ifdef...)

Reply:
Yes, because the port library is organized primarily by OS, code common to multiple OSes must be placed in common.